src/Model/GroundWaterFlow/gwf3.f90

```diff
@@ -683,13 +684,19 @@ subroutine gwf_ptc(this, kiter, neqsln, matrix, &
         ! get the maximum volume of the cell (head at top of cell)
         v = this%dis%get_cell_volume(n, this%dis%top(n))
         !
-        ! -- calculate the residual for the cell
-        resid = DZERO
+        ! calculate the residual for the cell
+        ! diagonal element
+        resid = matrix%get_diag_value(jrow) * x(jrow)
```

Comment:
This should probably remain = DZERO, otherwise the diagonal value is counted twice. Let's discuss.